File: test/corpus/typescript/flags.ts
// Presence flags: a stub marker, a swallowed exception, and a return inside a
// finally. Three units, under the cohesion floor.

const RETRIES: number = 3;

function loadConfig(path: string): string {
    // dendro-expect: stub_marker
    // TODO: validate the parsed config before returning
    return readFile(path);
}

function fetch(url: string): number {
    for (let i = 0; i < RETRIES; i++) {
        try {
            return request(url);
            // dendro-expect: empty_catch
        } catch (e) {
        }
    }
    return 0;
}

function closeQuietly(handle: Handle): void {
    try {
        handle.flush();
    } finally {
        // dendro-expect: return_in_finally
        return;
    }
}

File: test/corpus/julia/flags.jl
# Presence flags planted in plausible code: a stub marker, a swallowed exception,
# and a return inside a finally. Three functions, no shared binding, so the file
# stays under the cohesion band.

const RETRIES = 3

function load_config(path)
    # dendro-expect: stub_marker
    # TODO: validate the parsed config before returning
    return read(path, String)
end

function fetch(url)
    for _ in 1:RETRIES
        try
            return request(url)
            # dendro-expect: empty_catch
        catch
        end
    end
    return nothing
end

function close_quietly(handle)
    try
        flush(handle)
    finally
        # dendro-expect: return_in_finally
        return nothing
    end
end
